lib/class-dsl.php

```diff
@@ -158,16 +158,23 @@ public static function all_terms( string $field, array $values ): array {
 	 * Given DateTime objects for the from and to components of a range query,
 	 * builds the from/to components of the range query and returns it.
 	 *
-	 * @param DateTime $from The start date.
-	 * @param DateTime $to   The end date.
+	 * @param string|DateTime $from The start date.
+	 * @param string|DateTime $to   The end date.
```

Comment:
`DateTime::createFromFormat()` either returns a `DateTime` or `false`, it would never produce a string. Further, we'd definitely never want it to be a string.

I'd suggest that this be `DateTime|null`, the types in the function signature be `?DateTime`, and in `Custom_Date_Range::get_date_range()`, that call each argument as e.g. `$from_datetime ?: null`.